lib/types/gpt.nix

```diff
@@ -37,13 +37,18 @@ in
           device = lib.mkOption {
             type = lib.types.str;
             default =
-              if config._parent.type == "mdadm" then
-              # workaround because mdadm partlabel do not appear in /dev/disk/by-partlabel
+              if partition.config.uuid != null then
+                "/dev/disk/by-partuuid/${diskoLib.hexEscapeUdevSymlink partition.config.uuid}"
```

Comment:
do we need to hexescape here? Maybe it is better to add some checks to the type?

Reply:
I wasn't sure if there is a special way to indicate in a type, that it should be a string with a particular format. I'd be happy to include that if there is one.

Reply:
You can do that with `lib.types.strMatching` and a regex.  I'm surprised there's no predefined one for UUIDs.